communityApp/Screens/Signup.js

```diff
@@ -1,19 +1,53 @@
-import React, { useState } from 'react';
-import { View, Text, TextInput, TouchableOpacity, StyleSheet, Button} from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { View, ScrollView, Text, TextInput, TouchableOpacity, StyleSheet, Button} from 'react-native';
 import BlackButton from "./Components/BlackButton";
 import Header from './Components/Header';
 import Footer from './Components/Footer';
 import DateTimePickerModal from "react-native-modal-datetime-picker";
 import { SimpleLineIcons } from '@expo/vector-icons';
+import { createAccount } from '../API/Account';
+import { getInterestTypes } from '../API/Interest';
+import { getCommunities } from '../API/Community';
+import { SelectList, MultipleSelectList } from 'react-native-dropdown-select-list';
+import { validateEmail, validatePhone, validatePassword } from '../Utilities/AccountUtils';
 
 const SignupScreen = ({navigation}) => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [passwordConfirm, setPasswordConfirm] = useState('');
   const [name, setName] = useState('');
   const [number, setNumber] = useState('');
-  const [likes, setLikes] = useState('');
   const [birthday, setBirthday] = useState(new Date());
+  const [communitySelected, setCommunitySelected] = useState(null);
+  const [interestTypesSelected, setInterestTypesSelected] = useState([]);
+
+  // Some states that will be updated when the page is loaded.
+  const [communityAvailable, setCommunityAvailable] = useState([]);
+  const [interestTypesAvailable, setInterestTypesAvailable] = useState([]);
+
+  useEffect(() => {
```

Comment:
why is this a useeffect?

Reply:
It will fetch all the data from the database for interest types and community lists when the page is opened.